SimulationIOFilters/ImportDelamData.cpp

```diff
@@ -0,0 +1,568 @@
+
+#include <QtCore/QFileInfo>
+
+#include "SIMPLib/DataArrays/DataArray.hpp"
+#include "SIMPLib/DataContainers/DataContainerArray.h"
+#include "SIMPLib/FilterParameters/DataContainerCreationFilterParameter.h"
+#include "SIMPLib/FilterParameters/FloatFilterParameter.h"
+#include "SIMPLib/FilterParameters/InputFileFilterParameter.h"
+#include "SIMPLib/FilterParameters/StringFilterParameter.h"
+#include "SIMPLib/Geometry/RectGridGeom.h"
+
+#include "SimulationIO/SimulationIOConstants.h"
+#include "SimulationIO/SimulationIOVersion.h"
+
+enum createdPathID : RenameDataPath::DataID_t
+{
+  DataContainerID = 1,
+  AttributeMatrixID21 = 21,
+  DataArrayID30 = 30
+};
+
+// -----------------------------------------------------------------------------
+ImportDelamData::ImportDelamData()
+{
+  initialize();
+}
+
+// -----------------------------------------------------------------------------
+ImportDelamData::~ImportDelamData() = default;
+
+// -----------------------------------------------------------------------------
+void ImportDelamData::initialize()
+{
+  clearErrorCode();
+  clearWarningCode();
+  setCancel(false);
+}
+
+// -----------------------------------------------------------------------------
+void ImportDelamData::setupFilterParameters()
+{
+  FilterParameterVectorType parameters;
+
+  parameters.push_back(SIMPL_NEW_INPUT_FILE_FP("CSDGM File", CSDGMFile, FilterParameter::Parameter, ImportDelamData, "*.txt"));
+  parameters.push_back(SIMPL_NEW_INPUT_FILE_FP("Bvid StdOut File", BvidStdOutFile, FilterParameter::Parameter, ImportDelamData, "*.txt"));
+  parameters.push_back(SIMPL_NEW_FLOAT_FP("Interface Thickness", InterfaceThickness, FilterParameter::Parameter, ImportDelamData));
+
+  parameters.push_back(SIMPL_NEW_DC_CREATION_FP("Data Container", DataContainerName, FilterParameter::CreatedArray, ImportDelamData));
+  parameters.push_back(SIMPL_NEW_STRING_FP("Cell Attribute Matrix", CellAttributeMatrixName, FilterParameter::CreatedArray, ImportDelamData));
+  parameters.push_back(SIMPL_NEW_STRING_FP("Data Array", DataArrayName, FilterParameter::CreatedArray, ImportDelamData));
+
+  setFilterParameters(parameters);
+}
+
+// -----------------------------------------------------------------------------
+void ImportDelamData::dataCheck()
+{
+  clearErrorCode();
+  clearWarningCode();
+
+  QFileInfo fi(m_CSDGMFile);
+  if(m_CSDGMFile.isEmpty())
+  {
+    QString ss = QObject::tr("Bvid File is empty.  Please enter a file path.");
+    setErrorCondition(-2000, ss);
+    return;
+  }
+
+  if(!fi.isFile() || fi.suffix() != "txt")
+  {
+    QString ss = QObject::tr("Bvid File is not a text file.  Please enter the path to a text file.");
+    setErrorCondition(-2001, ss);
+    return;
+  }
+
+  fi.setFile(m_BvidStdOutFile);
+  if(m_BvidStdOutFile.isEmpty())
+  {
+    QString ss = QObject::tr("Bvid StdOut File is empty.  Please enter a file path.");
+    setErrorCondition(-2002, ss);
+    return;
+  }
+
+  if(!fi.isFile() || fi.completeSuffix() != "txt")
```

Comment:
```suggestion
  if(!fi.isFile() || fi.suffix().toLower() != "txt")
```

Changed to suffix().toLower() to match other check.